lib/optimizer/src/Transformer/ServerSideRendering.php

```diff
@@ -0,0 +1,628 @@
+            case Layout::FIXED_HEIGHT:
+                $styles = "height:{$height->getNumeral()}{$height->getUnit()};";
+                break;
+            case Layout::RESPONSIVE:
+            case Layout::INTRINSIC:
+                // Do nothing here but emit <i-amphtml-sizer> later.
+                break;
+            case Layout::FILL:
+            case Layout::CONTAINER:
+                // Do nothing here.
+                break;
+            case Layout::FLEX_ITEM:
+                if ($width->isDefined()) {
+                    $styles = "width:{$width->getNumeral()}{$width->getUnit()};";
+                }
+                if ($height->isDefined()) {
+                    $styles .= "height:{$height->getNumeral()}{$height->getUnit()};";
+                }
+                break;
+        }
+
+        // We prepend just in case an existing value (which shouldn't be there for valid docs) doesn't end with ';'.
+        if ($element->hasAttribute(Tag::STYLE)) {
+            $styles .= $element->getAttribute(Tag::STYLE);
+        }
+        if (! empty($styles)) {
+            $element->setAttribute(Tag::STYLE, $styles);
+        }
+
+        $element->setAttribute(Amp::LAYOUT_ATTRIBUTE, $layout);
+    }
+
+    /**
+     * Get the class to use for a given layout.
+     *
+     * @param string $layout Layout to get the class for.
+     * @return string Class name to use for the layout.
+     */
+    private function getLayoutClass($layout)
+    {
+        if (empty($layout)) {
+            return '';
+        }
+
+        return Amp::LAYOUT_CLASS_PREFIX . $layout;
+    }
+
+    /**
+     * Add a class to an element.
+     *
+     * This makes sure we keep existing classes on the element.
+     *
+     * @param DOMElement $element Element to add a class to.
+     * @param string     $class   Class to add.
+     */
+    private function addClass(DOMElement $element, $class)
+    {
+        if ($element->hasAttribute(Attribute::CLASS_) && ! empty($element->getAttribute(Attribute::CLASS_))) {
+            $class = "{$element->getAttribute(Attribute::CLASS_)} {$class}";
+        }
+
+        $element->setAttribute(Attribute::CLASS_, $class);
+    }
+
+    /**
+     * Check whether the provided layout is a layout with a defined size.
+     *
+     * @param string $layout Layout to check.
+     * @return bool Whether the layout has a defined size.
+     */
+    private function isLayoutSizeDefined($layout)
+    {
+        return in_array($layout, Layout::SIZE_DEFINED_LAYOUTS, true);
+    }
+
+    /**
+     * Insert a sizer element if one is required.
+     *
+     * @param Document   $document DOM document to add a sizer to.
+     * @param DOMElement $element  Element to add a sizer to.
+     * @param string     $layout   Calculated layout of the element.
+     * @param CssLength  $width    Calculated width of the element.
+     * @param CssLength  $height   Calculated height of the element.
+     */
+    private function maybeAddSizerInto(
```

Comment:
@amedina _He aquí_, `maybe` functions are not unique to WP 😄 https://github.com/ampproject/amp-toolbox/blob/236625a8bad997162c71ceb6cab351e634256552/packages/optimizer/lib/transformers/ApplyLayout.js#L93-L108